Shapes/rml_rules_shape.ttl

```diff
@@ -0,0 +1,1683 @@
+                    ns1:minCount 0 ;
+                    ns1:name "D2RQ username" ;
+                    ns1:nodeKind ns1:Literal ;
+                    ns1:path <http://www.wiwiss.fu-berlin.de/suhl/bizer/D2RQ/0.1#username> ],
+                [ ns1:datatype xsd:string ;
+                    ns1:maxCount 1 ;
+                    ns1:minCount 0 ;
+                    ns1:name "D2RQ password" ;
+                    ns1:nodeKind ns1:Literal ;
+                    ns1:path <http://www.wiwiss.fu-berlin.de/suhl/bizer/D2RQ/0.1#password> ] ],
+        [ ns1:description """
+
+        """ ;
+            ns1:in ( <http://www.w3.org/ns/r2rml#SQL2008> ) ;
+            ns1:maxCount 1 ;
+            ns1:message """
+        rr:sqlVersion must be rr:SQL2008. Providing rr:sqlVersion is optional.
+        """ ;
+            ns1:minCount 0 ;
+            ns1:name "rr:sqlVersion" ;
+            ns1:path <http://www.w3.org/ns/r2rml#sqlVersion> ],
+        [ ns1:description """
+        Exactly one rml:query or one rr:sqlQuery or one rr:tableName must be 
+        provided to access the relational database.
+        """ ;
+            ns1:maxCount 1 ;
+            ns1:minCount 1 ;
+            ns1:name "rml:query/rr:sqlQuery/rr:tableName" ;
+            ns1:path [ ns1:alternativePath ( <http://semweb.mmlab.be/ns/rml#query> <http://www.w3.org/ns/r2rml#sqlQuery> <http://www.w3.org/ns/r2rml#tableName> ) ] ],
+        [ ns1:datatype xsd:string ;
+            ns1:description """
+        An R2RML view is a logical table whose contents are the result of 
+        executing a SQL query against the input database. It is represented by 
+        a resource that has exactly one rr:sqlQuery property, whose value is a 
+        literal with a lexical form that is a valid SQL query.
+        """ ;
+            ns1:maxCount 1 ;
+            ns1:message """
+        rr:sqlQuery is required for querying a relational database. 
+        Must be provided once as a string.
+        """ ;
+            ns1:minCount 0 ;
+            ns1:name "rr:sqlQuery" ;
+            ns1:nodeKind ns1:Literal ;
+            ns1:path <http://www.w3.org/ns/r2rml#sqlQuery> ],
+        [ ns1:datatype xsd:string ;
+            ns1:description """
+        The query to execute on the data source.
+        rml:query is unofficially supported by RML, but not in the 
+        specification.
+        """ ;
+            ns1:maxCount 1 ;
+            ns1:message """
+        rml:query is required for relational database. 
+        Must be provided once as a string.
+        """ ;
+            ns1:minCount 0 ;
+            ns1:name "rml:query" ;
+            ns1:nodeKind ns1:Literal ;
+            ns1:path <http://semweb.mmlab.be/ns/rml#query> ],
+        [ ns1:datatype xsd:string ;
+            ns1:description """
+        The value of rr:tableName specifies the table or view name of 
+        the base table or view. Its value MUST be a valid 
+        schema-qualified name that names an existing base table or 
```

Comment:
Enhancement: there are rules for what is a valid SQL identifier (e.g., [for SQL92](https://docs.oracle.com/javadb/10.8.3.0/ref/crefsqlj1003454.html)), so there could be an `sh:pattern` checking that. But there might be vendor-specific exceptions to those rules, so maybe not worth the effort.

Reply:
Yeah indeed, vendor extensions are a mess :/ I would leave this open TBH.